09-maze_solver/progress/internal/solver/explore.go

```diff
@@ -52,42 +45,47 @@ func (s *Solver) explore(pathToBranch []image.Point) {
 		// We know we'll have up to 3 new neighbours to explore.
 		candidates := make([]image.Point, 0, 3)
 		for _, n := range neighbours(pos) {
-			if n == previous {
+			if pathToBranch.previousStep != nil && pathToBranch.previousStep.at == n {
```

Comment:
This could be a method: `isPreviousStep(n)`

Reply:
Done.